integration/query_test.go

```diff
@@ -1179,15 +1190,48 @@ func TestQueryCommandGetMore(t *testing.T) {
 				Message: "BSON field 'getMore.collection' is missing but a required field",
 			},
 		},
+		"UnsetAllBatchSize": {
+			findBatchSize:    nil,
+			getMoreBatchSize: nil,
+			collection:       collection.Name(),
+			firstBatch:       docs[:101],
+			nextBatch:        docs[101:],
+		},
+		"UnsetFindBatchSize": {
+			findBatchSize:    nil,
+			getMoreBatchSize: 5,
+			collection:       collection.Name(),
+			firstBatch:       docs[:101],
+			nextBatch:        docs[101:106],
+		},
+		"UnsetGetMoreBatchSize": {
+			findBatchSize:    5,
+			getMoreBatchSize: nil,
+			collection:       collection.Name(),
+			firstBatch:       docs[:5],
+			nextBatch:        docs[5:],
+		},
+		"BatchSize": {
+			findBatchSize:    3,
+			getMoreBatchSize: 5,
+			collection:       collection.Name(),
+			firstBatch:       docs[:3],
+			nextBatch:        docs[3:8],
+		},
 	} {
 		name, tc := name, tc
 		t.Run(name, func(t *testing.T) {
 			if tc.skip != "" {
 				t.Skip(tc.skip)
 			}
 
-			// TODO: https://github.com/FerretDB/FerretDB/issues/1807
-			// Do not run tests in parallel, MongoDB throws error that session and cursor do not match.
+			// Do not run subtests in t.Parallel() to reduce the occurrence
+			// of session error. Even the same connection is used for the test
+			// by setting `minPoolSize` and `maxPoolSize` to 1, on rare occasion
+			// it returns error due to different session calling `find` and `getMore`.
+			// Supporting session would help us understand fix it
+			// https://github.com/FerretDB/FerretDB/issues/153.
```

Comment:
So far this hasn't happened as long as sub tests don't run in parallel. But just a warning